### PR TITLE
backend: Upgrade langchain and fix /langchain-chat endpoint

### PR DESCRIPTION
**Pull request by tianjing-li**
Thank you for contributing to the Cohere Toolkit!

- [ ] **PR title**: "area: description"
  - Where "area" is whichever of interface, frontend, model, tools, backend, etc. is being modified. Use "docs: ..." for purely docs changes, "infra: ..." for CI changes.
  - Example: "deployment: add Azure model option"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change

- [ ] **Add tests and docs**: Please include testing and documentation for your changes
- [ ] **Lint and test**: Run `make lint` and `make run-tests`

**AI Description**

This PR updates the `langchain dependency to version `0.2.14 and its related packages. It also updates the `ruff` package to version `0.6.0.

## Package Updates
- `langchain` is updated to version `0.2.14.
- `langchain-community` is updated to version `0.2.12.
- `langchain-core` is updated to version `0.2.34.
- `langchain-text-splitters` is updated to version `0.2.2.
- `ruff` is updated to version `0.6.0.

## Code Changes
- Added imports from `logger` module in `langchain.py`.
- Added error logging in `langchain.py` when a tool is not found or does not have the `to_langchain_tool` method implemented.
- Updated the error message in `chat.py` to fix a typo.
- Added headers, send timeout, and ping to the `langchain_chat_stream` function in `chat.py`.
- Updated the comment and logic for extracting content from `action.message_log` in `chat.py`.
- Updated the comment for the shape of the content variable in `chat.py`.
- Removed the `documents` field from the `StreamToolResult` class in `chat.py`.
- Updated the final output logic in `chat.py` to use the `output` field directly and removed the unused `citations` and `documents` fields.

### CODE REVIEW

File: src/backend/chat/custom/langchain.py [hunk 1]
```diff
@@ -41,12 +44,16 @@ def chat(self, chat_request: LangchainChatRequest, **kwargs: Any) -> Any:
...
             tool = AVAILABLE_TOOLS.get(req_tool.name)
-            if tool:
+
+            if not tool:
+                logger.error(event=f"Failed to find tool {req_tool.name} during Langchain chat call.")
+
+            if not hasattr(tool.implementation, "to_langchain_tool"):
+                logger.error(event=f"Tool {req_tool.name} does not have to_langchain_tool method implemented.")
+
                 tools.append(tool.implementation().to_langchain_tool())
```
Comment (malexw) [inline src/backend/chat/custom/langchain.py:54]: Is the logic here correct? As it is now, it looks like the tool will only be appended if it doesn't have `to_langchain_tool` implemented. Do you want `continue` statements after each of the `logger.error` lines?

This also seems like a good area for a unit test.

Comment (tianjing-li) [reply src/backend/chat/custom/langchain.py]: You're right, fixed it by adding continues and moving the append code outside of the ifs (the latter error seems to be a copy paste issue)

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (tomtobac) [inline package-lock.json:1]: mmm did we push it unintentionally?

**Comment by tianjing-li** [comment]
No longer needed

**Pull request closed without merging**